lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs

```diff
@@ -357,7 +357,9 @@ type RecentEraLedgerConstraints era =
 -- https://cardanofoundation.atlassian.net/browse/ADP-2353
 withConstraints
     :: RecentEra era
-    -> ((RecentEraLedgerConstraints (CardanoApi.ShelleyLedgerEra era)) => a)
+    -> ( ( IsRecentEra era
+         , (RecentEraLedgerConstraints (CardanoApi.ShelleyLedgerEra era))
+         ) => a)
```

Comment:
This seems a bit awkward to read (and to format)!

Why not define a separate constraint synonym:
```hs
type RecentEraConstraints era =
    ( IsRecentEra era
    , RecentEraLedgerConstraints (CardanoApi.ShelleyLedgerEra era)
    )
```

Then we can write:
```hs
withConstraints :: RecentEra era -> (RecentEraConstraints era => a) -> a
```